_writeups/2023/ctf101/substitution-ciphers.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: writeup
+parent: CTF101
+grand_parent: 2023
+author: r98inver
+category: crypto
+tags: single-key-xor vigenere
+title: "Substitution Ciphers"
+subtitle: ""
+write_date: 2023-10-20
+last_edit_date:
+layout: mathjax # Uncomment this line to enable MathJax
+---
+
+The second crypto session of **CTF101** focused on symmetric enctryption, and more specifically substitution ciphers.
+
+## stellar-caesar
+
+The first challenge was a simple Caesar Cipher, as hinted by the name. The ciphertext is `vy{vkliwlqj_olnh_wkh_dqflhqw_urpdqv}` and just putting this in CyberChef with [ROT13 shifted by -3](https://gchq.github.io/CyberChef/#recipe=ROT13(true,true,false,-3)&input=dnl7dmtsaXdscWpfb2xuaF93a2hfZHFmbGhxd191cnBkcXZ9Cg) returns the flag: `sv{shifting_like_the_ancient_romans}`. The shift of -3 could be either computed by knowing that the flag begins with `sv`, or guessed, since it was the one supposedly used by Caesar, or even bruteforced easily.
+
+## trial-and-error
+
+The second challenge was the python version of the Caesar Cipher: a single byte XOR. The idea here is to simply bruteforce all the possible keys, since there are only 256 possibilities to chose one byte, and then find a way to detect when the output is correct (checking by hand 256 possibilities is still duable, but automating it is much more helpful for the next challenge and way less boring). First of all, we need a function to xor a key against our plaintext. We can use the `xor` function from `pwntools` or write a simple helper function like this:
```

Comment:
```suggestion
The second challenge was the python version of the Caesar Cipher: a single byte XOR. The idea here is to simply bruteforce all the possible keys, since there are only 256 possibilities to chose one byte, and then find a way to detect when the output is correct (checking by hand 256 possibilities is still doable, but automating it is much more helpful for the next challenge and way less boring). First of all, we need a function to xor a key against our plaintext. We can use the `xor` function from `pwntools` or write a simple helper function like this:
```